packages/blade/src/components/Tag/AnimatedTag.native.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import type { EasingFn } from 'react-native-reanimated';
+import Animated, {
+  Easing,
+  useSharedValue,
+  withTiming,
+  useAnimatedStyle,
+  runOnJS,
+  Keyframe,
+} from 'react-native-reanimated';
+import type { ReanimatedKeyframe } from 'react-native-reanimated/lib/typescript/reanimated2/layoutReanimation/animationBuilder/Keyframe';
+import { Tag } from './Tag';
+import {
+  TAG_MAX_WIDTH_END,
+  TAG_MAX_WIDTH_START,
+  TAG_OPACITY_END,
+  TAG_OPACITY_START,
+} from './tagAnimationConfig';
+import type { AnimatedTagProps } from './types';
+import { castNativeType, useTheme } from '~utils';
+
+const useAnimatedTag = (
+  onAnimationEndCallback: () => void,
+): {
+  entering: ReanimatedKeyframe;
+  animate: () => void;
+  animatedStyle: {
+    maxWidth: number;
+    opacity: number;
+  };
+} => {
+  const maxWidth = useSharedValue(TAG_MAX_WIDTH_START); // Initial max-width value
+  const opacity = useSharedValue(TAG_OPACITY_START); // Initial opacity value
+  const { theme } = useTheme();
+
+  const entering = new Keyframe({
+    0: {
+      opacity: TAG_OPACITY_END,
+    },
+    100: {
+      opacity: TAG_OPACITY_START,
+      easing: (theme.motion.easing.exit.effective as unknown) as EasingFn,
```

Comment:
use castNative

Reply:
castNative makes it EasingFactoryFn but here we want EasingFn.

Seems like we had to do same thing in other places too
https://github.com/razorpay/blade/blob/7c116d54164124651fe420a93f182fb4caf49cf9/packages/blade/src/components/Radio/RadioIcon/Fade.native.tsx#L20-L21